presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -310,7 +311,7 @@ public void onFailure(Throwable t)
                 {
                     finalMemoryFuture.set(null);
                 }
-            });
+            }, directExecutor());
```

Comment:
Replace the above with `memoryPoolFuture.addListener()` since both methods are the same.